website/docs/r/nsxv_dhcp_relay.markdown

```diff
@@ -0,0 +1,97 @@
+DHCP infrastructure. DHCP messages are relayed from virtual machines to the designated DHCP servers
+in your physical DHCP infrastructure, which allows IP addresses controlled by the NSX software to
+continue to be in synch with IP addresses in the rest of your DHCP-controlled environments. 
+
+Supported in provider *v2.6+*
+
+## Example Usage 1 (Minimal configuration)
+
+```hcl
+resource "vcd_nsxv_dhcp_relay" "relay_config" {
+  org          = "my-org"
+  vdc          = "my-org-vdc"
+  edge_gateway = "my-edge-gw"
+
+  ip_addresses = ["1.1.1.1"]
+
+  relay_agent {
+    org_network = vcd_network_routed.test-routed[0].name
+  }
+}
+```
+
+## Example Usage 2 (Example of configuration with multiple relay agents)
+
+```hcl
+resource "vcd_nsxv_dhcp_relay" "relay_config" {
+  org          = "my-org"
+  vdc          = "my-org-vdc"
+  edge_gateway = "my-edge-gw"
+
+  ip_addresses = ["1.1.1.1", "2.2.2.2"]
+  domain_names = ["servergroups.domainname.com", "other.domain.com"]
+  ip_sets      = [vcd_ipset.myset1.name, vcd_ipset.myset2.name]
+
+  relay_agent {
+    org_network = "my-routed-network-1"
+  }
+
+  relay_agent {
+    org_network        = vcd_network_routed.db-network.name
+    gateway_ip_address = "10.201.1.1"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level.
+* `edge_gateway` - (Required) The name of the edge gateway on which DHCP relay is to be configured.
+* `ip_addresses` - (Optional) A set of IP addresses.
+* `domain_names` - (Optional) A set of domain names.
+* `ip_sets` - (Optional) A set of IP set names.
+* `relay_agent` - (Required) One or more blocks to define Org network and optional IP address of
+  edge gateway interfaces from which DHCP messages are to be relayed to the external DHCP relay
+  server(s). See [Relay Agent](#relay-agent) and example for usage details.
+
+<a id="relay-agent"></a>
+## Relay Agent
+
+* `org_network` - (Required) An existing Org network name from which DHCP messages are to be relayed.
+* `gateway_ip_address` - (Optional) IP address on edge gateway to be used for relaying messages.
+  Primary address of edge gateway interface will be picked if not specified. 
+
+## Importing
+
+~> **Note:** The current implementation of Terraform import can only import resources into the state.
+It does not generate configuration. [More information.](https://www.terraform.io/docs/import/)
+
+An existing DHCP relay configuration can be [imported][docs-import] into this resource
+via supplying the full dot separated path for your edge gateway. An example is
+below:
+
+[docs-import]: https://www.terraform.io/docs/import/
+
+```
+terraform import vcd_nsxv_dhcp_relay.imported my-org.my-org-vdc.my-edge-gw
```

Comment:
This raises thoughts. So, we have 1:1 mapping between DHCP relay resource and an EG.
- Would be useful to note in docs that only a single DHCP relay can be set for a single EG.
- What would happen if user defined two relay blocks for a single EG?

Reply:
> This raises thoughts. So, we have 1:1 mapping between DHCP relay resource and an EG.
> 
>     * Would be useful to note in docs that only a single DHCP relay can be set for a single EG.
I had thoughts to note it is a "singleton". Probably worth it if as now we both have this idea.
> 
>     * What would happen if user defined two relay blocks for a single EG?
They would "fight" each other. Latest would win. Next apply would propose to "update"...

Reply:
> I had thoughts to note it is a "singleton". Probably worth it if as now we both have this idea.
Sounds good, of course you'll need to define scope to be on EG name.

Reply:
Added a note